File: todo-app-flutter-backend/src/models/todo.ts
import { model, Schema } from "mongoose";
import { Todo } from "../types/types";


const todoSchema = new Schema<Todo>({
    id: { type: String },
    title: { type: String, required: true },
    isFinished: { type: Boolean, required: true }
});

const TodoModel = model<Todo>('Todo', todoSchema);

export default TodoModel;

File: lib/models/todo.dart
class Todo {
  String? id;
  String title = '';
  bool isFinished = false;
  String? userId;
  Todo({this.id, required this.title, this.isFinished = false, this.userId});

  Map<String, dynamic> toJson() {
    return {'title': title, 'isFinished': isFinished};
  }

  factory Todo.fromJson(Map<String, dynamic> json) {
    return Todo(
      id: json['_id'],
      title: json['title'],
      isFinished: json['isFinished'],
    );
  }
}
